src/adapters/http/index.ts

```diff
@@ -0,0 +1,109 @@
+import Adapter from '../../lib/adapter.js';
+import GleeMessage from '../../lib/message.js';
+import http from 'http';
+import { validateData } from '../../lib/util.js';
+import GleeError from '../../errors/glee-error.js';
+import * as url from 'url';
+
+class HttpAdapter extends Adapter {
+    private res: any
+
+    name(): string {
+        return 'HTTP server'
+    }
+
+    async connect(): Promise<this> {
+        return this._connect();
+    }
+
+    async send(message: GleeMessage): Promise<void> {
+        console.log('control reached send');
+        return this._send(message);
+    }
+
+    _connect(): Promise<this> {
+        return new Promise((resolve, reject) => {
+            const serverUrl = new URL(this.serverUrlExpanded);
+            const httpServer = this.glee.options?.websocket?.httpServer || http.createServer();
+            const asyncapiServerPort = serverUrl.port || 80;
+            const optionsPort = this.glee.options?.websocket?.port;
+            const port = optionsPort || asyncapiServerPort;
+
+            httpServer.on('request', (req, res) => {
+                this.res = res;
+                let { pathname } = new URL(req.url, serverUrl);
+
+                if (pathname.startsWith('/')) {
+                    pathname = pathname.substring(1);
+                }
+
+                if (!this.parsedAsyncAPI.channel(pathname)) {
+                    res.end('HTTP/1.1 404 Not Found\r\n\r\n');
+                    const err = new Error(`A client attempted to connect to channel ${pathname} but this channel is not defined in your AsyncAPI file.`)
+                    this.emit('error', err)
+                    return reject(err)
+                }
+
+                const {query: searchParams} = url.parse(req.url, true);
+                const httpChannelBinding = this.parsedAsyncAPI.channel(pathname).binding('http');
+
+                if (httpChannelBinding) {
+                    const { query, method } = httpChannelBinding;
+
+                    if(method) {
+                        if(req.method !== method) {
+                            this.emit('error', new Error(`${req.method} is not allowed for this url`));
```

Comment:
What about the typical error message:

```suggestion
                            this.emit('error', new Error(`Cannot ${req.method} ${pathname}`));
```

This should give the user something like `Cannot POST /test`.